subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/DependencyHandlerScope.kt

```diff
@@ -21,21 +21,27 @@ import org.gradle.api.artifacts.Dependency
 import org.gradle.api.artifacts.ExternalModuleDependency
 import org.gradle.api.artifacts.ModuleDependency
 import org.gradle.api.artifacts.dsl.DependencyHandler
+import org.gradle.api.plugins.ExtensionAware
 
 
 /**
  * Receiver for `dependencies` block providing convenient utilities for configuring dependencies.
  *
  * @see [DependencyHandler]
  */
-class DependencyHandlerScope
+open class DependencyHandlerScope
 private constructor(
     val dependencies: DependencyHandler
 ) : DependencyHandler by dependencies {
 
     companion object {
         fun of(dependencies: DependencyHandler) =
-            DependencyHandlerScope(dependencies)
+            ExtensionAwareDependencyHandlerScope(dependencies) as DependencyHandlerScope
+
+        private
+        class ExtensionAwareDependencyHandlerScope(
+            dependencies: DependencyHandler
+        ) : DependencyHandlerScope(dependencies), ExtensionAware by (dependencies as ExtensionAware)
```

Comment:
Interesting way of handling this. Can you elaborate as to why you chose to go this route instead of adding `ExtensionAware` to `DependencyHandlerScope`?
By doing this, users in their build scripts can't access the rest of the methods that come along with having a type be `ExtensionAware`. For example:
```kotlin
dependencies {
    val someStoredString: String by this
}
```

Just trying to understand the thought process here.

Reply:
Adding `ExtensionAware` to `DependencyHandlerScope` would bring non-uniformity and confusion because it would not be available on `DependencyHandler. Taking one step at a time.

Reply:
Fair enough.